syntax = "proto3";

option go_package = "services/proto";

message LikeUndoDetails {
  int64 article_id = 1;
  string liker_handle = 2;
}

message UndoResponse {
  enum ResultType {
    OK = 0;
    ERROR = 1;
  }

  ResultType result_type = 1;

  // Should only be set if result_type is not OK
  string error = 2;
}

message ReceivedLikeUndoDetails {
  // The ActivityPub ID of the object being liked.
  string liked_object_ap_id = 1;

  // The ActivityPub ID of the user liking the object.
  string liking_user_ap_id = 2;
}


// Service for sending and receiving server-to-server undo activities.
service S2SUndo {
  rpc ReceiveLikeUndoActivity(ReceivedLikeUndoDetails) returns (UndoResponse);
  rpc SendLikeUndoActivity(LikeUndoDetails) returns (UndoResponse);
  // Further specialisations, i.e. SendArticleUndo, to be placed here.
}
